src/providers/yugen.rs

```diff
@@ -0,0 +1,72 @@
+use regex::Regex;
+use reqwest::{Client, Response};
+use reqwest_middleware::ClientWithMiddleware;
+use scraper::{Html, Selector, element_ref::Select};
+
+use crate::types::{AnimeEpisode, SearchResult, StreamLink};
+
+const host: &str = "https://yugen.to";
```

Comment:
The host should be just the hostname, nothing else.

(This might be used for picking the provider by url in the future)